students/emorychristenson/lesson04/mailroom_pt2.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Collection of existing donor data
+donors = {
+    "John Jacob": [5.00, 1000.00, 4.25],
+    "Jingleheimer Schmidt": [240.50],
+    "Chanandler Bong": [15000.00, 4000.00],
+    "Jimni Christmas": [12.00],
+    "S. Claus": [7000.00, 14000.00, 200.00]
+}
+
+# Message that displays when script is run
+main_prompt = """
+Please choose from the below options.\n
+1 - Send a thank you
+2 - Create a report
+3 - Send thanks to all donors
+4 - Quit\n
+"""
+def thanks_letter(donor, amount):
+    letter = (f"""\n
+    Dear {donor},
+
+    Thank you for your donation(s) of {amount:.2f}!
+    We really appreciate your support!
+    
+    Sincerely,
+        The Black Mesa Research Foundation""")
+    return letter
+
+# Allows user to select existing donor, add a new donor, or list all donors from the collection
+def thank_you():
+    print("\nSend a thank you!\n")
+    response = input("""Please type the name of the donor you'd like to thank.\n
+Use 'list' to view current donors, or 'quit' to return to the menu: """)
+    while response.lower() == "list":
+        for name in donors.keys():
+            print("\n",name)
+        response = input("\nPlease enter a donor name: ")
+    if response.lower() == "quit":
+        main()
+    else:
+        amount = float(input("\nEnter a donation amount: "))
+        
+    for donor in donors.keys():
+        if response.lower() == donor.lower():
+            donors[donor].append(amount)
+        else:        
+            donors[response] = [amount]
+        send_email(donor, amount)
```

Comment:
with dictionaries you should be checking if a key exists, instead of looping over keys:


```
if donors.get(donor):
     # append amount
else:
    # add new donor
```